lib/ansible/modules/crypto/crypt_device.py

```diff
@@ -0,0 +1,446 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: crypt_device
+
+short_description: Manage encrypted (LUKS) devices
+
+version_added: "2.8"
+
+description:
+    - "Module manages LUKS on given device. Supports creating, destroying,
+      opening and closing of LUKS container and adding or removing new keys."
+
+options:
+    device:
+        description:
+            - "Device to work with (e.g. '/dev/sda1'). Needed in most cases."
+        type: str
+    state:
+        description:
+            - "Based on its value creates or destroys the LUKS container on
+              a given device. Container creation requires 'device' and 'key'
+              options to be supplied."
+        default: present
+        choices: ['absent', 'present']
+        type: str
+    open:
+        description:
+            - "Based on its value unlocks or locks an existing LUKS container
+              on a given device. Use the 'name' option to set the name of
+              the opened container. Otherwise the name will be generated
+              automatically and returned in the result."
+        default: false
+        type: bool
+    name:
+        description:
+            - "Sets container name when 'open' is true. Can be used
+              instead of 'device' when closing the container
+              (i.e. when 'open' is set to false)."
+        type: str
+    key:
+        description:
+            - "Used to unlock the container and needed for most of the operations.
+              Parameter value is the path to keyfile with the passphrase.
+              BEWARE that working with keyfiles in plaintext is dangerous.
+              Use Ansible Vault to protect them."
```

Comment:
Removing the Vault part is also OK; and the `key_path` / `key_content` thing can also be implemented in another PR.

Reply:
After some thinking I have decided to not mention Vault at all. At the moment I am not sure about security of usage `key_content`.

Reply:
Adding `key_content` would also require adding `new_key_content` and `remove_key_content` options to stay coherent. That would lead to much more complex evaluation logic. I do agree with you that it would help to have this option but I would rather make it a future enhancement.

Reply:
After giving it some thought: It should be possible to check whether supplied string is a path to some existing file or not. Based on this check the `key` option can accept both: passphrase and path to a keyfile and then resolve it automatically. Still, I would rather add it in a future when this version proves its worth.

Reply:
Yes, it would also need `new_key_content` and `remove_key_content` at some point. And yes, let's do stuff like this in the future :)

About the idea of `key` doing some kind of detection magic: I don't think this is a good idea. Someone might actually use pathnames (or something which looks like a pathname) as a passphrase (maybe because it's easier to memorize and fulfils some "must contain other character" condition?). From experience, sooner or later this always becomes a problem, and once we are at that point, it's really hard to fix it without breaking backwards compatibility.

I think just adding `_path` to all these options is probably the best thing right now, even just because it makes it less likely that a user accidentally puts the key content itself in there (and then it's in the logs, since paths usually aren't marked as `no_log`).

Reply:
Danger of getting passphrases into logs by mistake is a strong argument. I have changed option names to `keyfile`, `new_keyfile` and `remove_keyfile`.